src/app/modules/wallet/pages/account-voluntary-exit/account-voluntary-exit.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Component, OnInit } from '@angular/core';
+import { BaseComponent } from '../../../shared/components/base.component';
+import { WalletService } from '../../../core/services/wallet.service';
+import { map } from 'rxjs/operators';
+import {
+  Account,
+  AccountVoluntaryExitRequest,
+} from '../../../../proto/validator/accounts/v2/web_api';
+import {
+  FormGroup,
+  FormBuilder,
+  Validators,
+  AbstractControlOptions,
+} from '@angular/forms';
+import { UtilityValidator } from '../../../onboarding/validators/utility.validator';
+import { ActivatedRoute } from '@angular/router';
+import { base64ToHex } from 'src/app/modules/core/utils/hex-util';
+import { MatSelectionListChange } from '@angular/material/list';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+
+@Component({
+  selector: 'app-account-voluntary-exit',
+  templateUrl: './account-voluntary-exit.component.html',
+  styleUrls: ['./account-voluntary-exit.component.scss']
+})
+export class AccountVoluntaryExitComponent   extends BaseComponent
+implements OnInit {
+
+
+  constructor(
+    private walletService: WalletService,
+    private activatedRoute: ActivatedRoute,
+    private snackMsgService: MatSnackBar,
+    private formBuilder: FormBuilder
+  ) {
+    super();
+  }
+  exitAccountFormGroup: FormGroup | undefined | null;
+  keys: Account[] = [];
+  ngOnInit(): void {
+    const publicKey = this.activatedRoute.snapshot.queryParams.publicKey;
+    this.exitAccountFormGroup = this.formBuilder.group(
+      {
+        confirmation: [
+          '',
+          [Validators.required, UtilityValidator.MustBe('agree')],
+        ],
+      },
+      {
+        validators: UtilityValidator.MustHaveMoreThanOneControle,
+      } as AbstractControlOptions
+    );
+
+    this.walletService
```

Comment:
It's possible we want to exit N public keys where N is anything from 1 to the total number of keys in the wallet, inclusive. Maybe we can input a list of public keys into this component and then you wont need to query the wallet service